src/OSLikeStuff/task_scheduler.cpp

```diff
@@ -386,6 +387,9 @@ void TaskManager::start(double duration) {
 	if (duration != 0) {
 		mustEndBefore = startTime + duration;
 	}
+	else {
+		mustEndBefore = -1;
```

Comment:
I thought it was weird to ignore the `duration` parameter if it is non-zero, it will not work if you call `start()` first with a timeout and later without a timeout.

Reply:
hm true